### PR TITLE
Blackstone Port - Faith Overhaul + Misc Patron Fixes

### PR DESCRIPTION
**Pull request by Stutternov**
## About The Pull Request

**Tl;dr** - I ported the PR listed below and made changes to it to make it align to our current god's lore. I know it's changing and shit, but hey, get patrons working and stuff. I don't take credit for really any of the codework here other than just changing shit around where needed, the author of Blackstone's PR did **ALL** of the work and I'm just porting it here. Man is doing Gods work, literally.

https://github.com/Blackstone-SS13/BLACKSTONE/pull/756

**Ok but R-Rebel, what it do???**
- Makes patron gods selectable.
- Assigns evil gods to antags, i.e Zizo for vampires, Matthios for bandits, etc.
- Neatens up the age restrictions, race restrictions, and patrons jobs can have.
- Basically future-proofs if we want more than one faith itself, such as some other race's god stuff, more antag gods, etc.
- Future-proofs and streamlines god devotion abilities; I didn't add or change any but makes it much easier to go through and add/remove them.

Incorporated lore added in https://github.com/Darkrp-community/OpenKeep/pull/207

## Why It's Good For The Game

Fixes a bug we have had for ages and streamlines patron selection, making it easier to change and/or add to in the future while providing even better flavor to players.

## Pre-Merge Checklist

- [X] You tested this on a local server.
- [X] This code did not runtime during testing.
- [X] You documented all of your changes.

### CODE REVIEW

**Comment by BadAtThisGame302** [comment]
Should make the Court Mage Noc Patron only.

**Comment by Stutternov** [comment]
Right, tested this. It appears to actually properly remember your selected patron now, it allows you to select faith choice (but there's only one right now, divine pantheon), and jobs **are** properly locked.

The only 'issue' I see is, if you don't worship Astra on your character, you cannot join as a Priest. It locks jobs based on your diety. Which, isn't an issue per-say, but I'm not sure if that's wanted or not. I think it's fine as long as it gets outlined in the role description.

What I need now is just to be told what patrons any specific roles should be locked behind.

Example, Astra is the only deity priest can have, clergy can have a handful of specific ones, and Court Mage can only have Noc.

**Comment by Stutternov** [comment]
Updated & merged in lore changes asked for.

**Comment by Stutternov** [comment]
Tested it again today, still working. Merge conflicts were resolved.

**Comment by Stutternov** [comment]
> Update and testmerge IMO, its a core system thats needed.

Updated, testmerging will be tomorrow when I'll be free to emergency unmerge incase of unforeseen breaking. This was really not designed with the drifter system so I fear how it'll interact with it.
